TrigScint/src/TrigScint/TrigScintFirmwareHitProducer.cxx

```diff
@@ -0,0 +1,97 @@
+
+#include "TrigScint/TrigScintFirmwareHitProducer.h"
+
+#include <iterator>
+#include <map>
+
+#include "TrigScint/Firmware/hitproducer.h"
+#include "TrigScint/Firmware/objdef.h"
+
+namespace trigscint {
+
+void TrigScintFirmwareHitProducer::configure(
+    framework::config::Parameters &ps) {
+  pedestal_ = ps.getParameter<double>("pedestal");
+  gain_ = ps.getParameter<double>("gain");
+  mevPerMip_ = ps.getParameter<double>("mev_per_mip");
+  pePerMip_ = ps.getParameter<double>("pe_per_mip");
+  inputCollection_ = ps.getParameter<std::string>("input_collection");
+  testCollection_ = ps.getParameter<std::string>("test_collection");
+  doTest_ = ps.getParameter<bool>("do_test");
+  inputPassName_ = ps.getParameter<std::string>("input_pass_name");
+  outputCollection_ = ps.getParameter<std::string>("output_collection");
+  verbose_ = ps.getParameter<bool>("verbose");
+  sample_of_interest_ = ps.getParameter<int>("sample_of_interest");
+  if (verbose_) {
+    ldmx_log(info) << "In TrigScintFirmwareHitProducer: configure done!";
+    ldmx_log(info) << "\nPedestal: " << pedestal_ << "\nGain: " << gain_
+                   << "\nMEV per MIP: " << mevPerMip_
+                   << "\nPE per MIP: " << pePerMip_
+                   << "\ninput collection:     " << inputCollection_
+                   << "\ntest collection:	" << testCollection_
+                   << "\nAre we testing:        " << doTest_
+                   << "\nInput pass name:     " << inputPassName_
+                   << "\nOutput collection:    " << outputCollection_
+                   << "\nVerbosity: " << verbose_;
+  }
+
+  return;
+}
+
+void TrigScintFirmwareHitProducer::produce(framework::Event &event) {
+  if (doTest_) {
+    const auto rechits{event.getCollection<ldmx::TrigScintHit>(testCollection_,
+                                                               inputPassName_)};
+    for (const auto &hit : rechits) {
+      std::cout << "Analysis barID: " << hit.getBarID()
```

Comment:
Move this to the logger system

Reply:
done I think? You mean use verbose, correct?

Reply:
no no I meant 
```
ldmx_log(debug)
```
and then you dont need the `<< std::endl` either

Reply:
Gotcha! Implemented it :)